aten/src/ATen/native/mkldnn/xpu/detail/Utils.h

```diff
@@ -0,0 +1,360 @@
+#pragma once
+#include <iostream>
+#include <ATen/ATen.h>
+#include <ATen/Tensor.h>
+#include <ATen/core/Tensor.h>
+
+#include <ATen/core/grad_mode.h>
+#include <c10/core/MemoryFormat.h>
+#include <oneapi/dnnl/dnnl.hpp>
+#include <oneapi/dnnl/dnnl_sycl.hpp>
+#include <oneapi/dnnl/dnnl_version.h>
+
+
+#define ONEDNN_SUPPORT_DETERMINISTIC (DNNL_VERSION_MAJOR >=3 && DNNL_VERSION_MINOR >=4)
+
+#define XPU_ONEDNN_EXEC(prim, stream, ...)                               \
+{                                                                        \
+  auto e = dnnl::sycl_interop::execute((prim), (stream), ##__VA_ARGS__); \
```

Comment:
As long as the e is created in {}, it will be useless for rest of op impl, right?

Reply:
Yes, I would remove this marco and directly call execute in primitive. I think this marco is not needed anymore.

Reply:
hi, @gujinghui , I have removed this marco, the primitive would directly use `dnnl::sycl_interop::execute` for calling primitive. 

Also, I have change the matmul and convolution API, which would expose `sycl::event` to user just like the codes in IPEX.  You may refer to 

https://github.com/pytorch/pytorch/pull/117112/files#diff-b3a3b45f243b153b61b63b33bee989ef2ecd7f5b28970555adfc7d4a92b35b81R11, 

and conv in #117512

Reply:
Looks great. BTW, will you append below APIs into matmul integration layer?
https://github.com/intel-innersource/frameworks.ai.pytorch.ipex-gpu/pull/3926

Reply:
Yes, I may put it in another separate PR, current PRs would focus on the high priority ops, like addbmm, bmm.